skyplane/obj_store/azure_blob_interface.py

```diff
@@ -149,28 +150,100 @@ def download_object(
+        :type dst_object_name: str
+        :param mime_type: unused in this function but is kept for consistency with the other interfaces (default: None)
+        :type mime_type: str
+        """
+
+        assert len(dst_object_name) > 0, f"Destination object name must be non-empty: '{dst_object_name}'"
+
+        return dst_object_name
+
+    @imports.inject("azure.storage.blob", pip_extra="azure")
+    def complete_multipart_upload(azure_blob, self, dst_object_name: str, upload_id: str, custom_data: Optional[Any] = None) -> None:
+        """After all blocks of a blob are uploaded/staged with their unique block_id,
+        in order to complete the multipart upload, we commit them together.
+
+        :param dst_object_name: name of the destination object, also is used to index into our block mappings
+        :type dst_object_name: str
+        :param upload_id: upload_id to index into our block id mappings, should be the same as the dst_object_name in Azure
+        :type upload_id: str
+        :param custom_data: In Azure, this custom data is the blockID list (parts) and the object mime_type from the TransferJob instance (default: None)
+        :type custom_data: Optional[Any]
+        """
+
+        assert upload_id == dst_object_name, "In Azure, upload_id should be the same as the blob name."
+        assert custom_data is not None, "In Azure, the custom data should exist for multipart"
+
+        # Decouple the custom data
+        block_list, mime_type = custom_data
+        assert block_list != [], "The blockID list shouldn't be empty for Azure multipart"
+        block_list = list(map(lambda block_id: azure_blob.BlobBlock(block_id=block_id), block_list))
+
+        blob_client = self.blob_service_client.get_blob_client(container=self.container_name, blob=dst_object_name)
+        try:
+            # The below operation will create the blob from the uploaded blocks.
+            blob_client.commit_block_list(block_list=block_list, content_settings=azure_blob.ContentSettings(content_type=mime_type))
+        except Exception as e:
+            raise exceptions.SkyplaneException(f"Failed to complete multipart upload for {dst_object_name}: {str(e)}")
+
+    @staticmethod
+    def id_to_base64_encoding(part_number: int) -> str:
+        """Azure expects all blockIDs to be Base64 strings. This function serves to convert the part numbers to
+        base64-encoded strings. Used within upload_object
+
+        :param part_number: part number of the block, determined while splitting the date into chunks before the transfer
+        :type part_number: int
+        """
+        block_id = format(part_number, "06")  # pad with zeros to get consistent length
+        block_id = block_id.encode("utf-8")
+        block_id = base64.b64encode(block_id).decode("utf-8")
+        return block_id
+
     def create_object_repr(self, key: str) -> AzureBlobObject:
+        self.provider = "azure"

```

Comment:
I am not sure if this was an error, but I set this here manually since I was getting: `'AzureBlobInterface' object has no attribute 'provider'` error

Reply:
Removed this on the most-up-to-date version

Reply:
This is fixed now